dlt/destinations/impl/clickhouse/sql_client.py

```diff
@@ -98,13 +103,16 @@ def execute_sql(
             return None if curr.description is None else curr.fetchall()
 
     def create_dataset(self) -> None:
-        # We create a sentinel table which defines wether we consider the dataset created
+        # We create a sentinel table which defines whether we consider the dataset created.
         sentinel_table_name = self.make_qualified_table_name(
             self.credentials.dataset_sentinel_table_name
         )
-        self.execute_sql(
-            f"""CREATE TABLE {sentinel_table_name} (_dlt_id String NOT NULL PRIMARY KEY) ENGINE=ReplicatedMergeTree COMMENT 'internal dlt sentinel table'"""
-        )
+        # `MergeTree` is guaranteed to work in both self-managed and cloud setups.
+        self.execute_sql(f"""
+            CREATE TABLE {sentinel_table_name}
+            (_dlt_id String NOT NULL PRIMARY KEY)
+            ENGINE=MergeTree
```

Comment:
I'm probably nitpicking but should it use config.table_engine_type here?

Reply:
@rudolfix Thanks for the spot!

Changed to config table engine:

https://github.com/dlt-hub/dlt/blob/9c67c65cc1e50fb4ab746bebf0437485b600fff0/dlt/destinations/impl/clickhouse/sql_client.py#L116-L121